removed help and account from mobile, fixed anchor nesting

KatarzynaStefanowiczPagepro opened the pull request:
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=776) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/JoinColony/colonyCDapp?pullRequest=776) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/JoinColony/colonyCDapp?pullRequest=776) it.</sub>

KatarzynaStefanowiczPagepro commented:
https://tw.auroracreation.com/app/tasks/15516455

![image](https://github.com/JoinColony/colonyCDapp/assets/119701775/6bc7ac51-a083-4cf7-ad14-e991e7799cd1)

- removed Help and account text from mobile
- fixed nesting anchors
